pkg/sql/ir/ir.go

```diff
@@ -76,17 +76,34 @@ type Executor interface {
 type SQLStatement interface {
 	SetOriginalSQL(string)
 	Execute(Executor) error
+	IsExtended() bool
+	GetOriginalSQL() string
 }
 
-// TrainStmt is the intermediate representation for code generation of a training job.
-type TrainStmt struct {
+// ExtendedSQL contains common fields of TrainStmt, PredictStmt and AnalyzeStmt
+type ExtendedSQL struct {
```

Comment:
The `ExtendedSQL` represent IR, do you want to rename `ExtendedSQL ` to `ExtendedStatment`